astropy/coordinates/tests/test_earth.py

```diff
@@ -285,35 +285,47 @@ def test_repr_latex():
 
 
 @pytest.mark.remote_data
-def test_of_address():
+# TODO: this parametrize should include a second option with a valid Google API
+# key. For example, we should make an API key for Astropy, and add it to Travis
+# as an environment variable (for security).
+@pytest.mark.parametrize('google_api_key', [None])
+def test_of_address(google_api_key):
+    NYC_lon = -74.0 * u.deg
+    NYC_lat = 40.7 * u.deg
+    # ~10 km tolerance to address difference between OpenStreetMap and Google
+    NYC_tol = 0.1 * u.deg
+
     # just a location
     try:
         loc = EarthLocation.of_address("New York, NY")
     except NameResolveError as e:
-        # Google map API limit might surface even here in Travis CI.
-        pytest.xfail(str(e))
+        # API limit might surface even here in Travis CI.
+        if 'unknown failure with' not in str(e):
+            pytest.xfail(str(e))
     else:
-        assert quantity_allclose(loc.lat, 40.7128*u.degree)
-        assert quantity_allclose(loc.lon, -74.0059*u.degree)
+        # Tolerance below is a MAGIC NUMBER
```

Comment:
Maybe just say a bit more that this magic is because google and OSM don't give the same answer?

Reply:
I just removed this line, because above there is this comment:
`# ~10 km tolerance to address difference between OpenStreetMap and Google`

Reply:
Would providing a street address decrease the difference? Or maybe there is a reason why a vague "New York, NY" was used?

Reply:
I am just thinking that a 10 km difference would be scary if I were to use it for navigation, but then I realized "center of a city" can be quite subjective. As a user, often times one wants exact location of a given point (say, observatory); in that case, 10 km difference is unacceptable?

Reply:
So in other words, are we testing this properly?

Reply:
This isn't testing the accuracy of the returned earth location, think of it as a unit test that the query is actually succeeding. It's out of scope of Astropy to verify that the latitude/longitude is "correct" -- the docstring clearly states that "This is intended as a quick convenience function to get fast access to locations." I'll add the additional line to say that if you require precision in earth locations, the user should pass in their own! I wouldn't be surprised if the *addresses* of observatories are only good to within a km or so...

Reply:
Okay, fair enough. Thanks for the clarification!